test/controllers/overseas.name.controller.spec.ts

```diff
@@ -195,24 +211,42 @@ describe("Overseas Name controller", () => {
       expect(resp.text).toContain(`${FOUND_REDIRECT_TO} ${PRESENTER_URL}`);
     });
 
-    test(`redirect to the ${PRESENTER_PAGE} page after a successful creation of transaction and overseas entity`, async () => {
+    test(`redirect to the ${PRESENTER_PAGE} page after a successful creation of transaction and overseas entity with REDIS_removal flag set to OFF`, async () => {
       mockIsActiveFeature.mockReturnValueOnce(true); // For FEATURE_FLAG_ENABLE_SAVE_AND_RESUME
+      mockIsActiveFeature.mockReturnValueOnce(false); // For FEATURE_FLAG_ENABLE_REDIS_REMOVAL
 
-      const mockData = { ...APPLICATION_DATA_MOCK, [Transactionkey]: "", [OverseasEntityKey]: "" };
-      mockGetApplicationData.mockReturnValueOnce({}); // Needed at the setExtraData
-      mockGetApplicationData.mockReturnValueOnce(mockData); // Needed inside the feature flag
+      const mockData = { ...APPLICATION_DATA_MOCK, [Transactionkey]: null, [OverseasEntityKey]: null };
+      mockGetApplicationData.mockReturnValueOnce(mockData);
       const resp = await request(app).post(OVERSEAS_NAME_URL).send({ [EntityNameKey]: OVERSEAS_NAME_MOCK });
 
       expect(resp.status).toEqual(302);
-      expect(mockData[Transactionkey]).toEqual(TRANSACTION_ID);
-      expect(mockData[OverseasEntityKey]).toEqual(OVERSEAS_ENTITY_ID);
+      expect(mockData[Transactionkey]).toEqual(null);
+      expect(mockData[OverseasEntityKey]).toEqual(null);
       expect(mockTransactionService).toHaveBeenCalledTimes(1);
       expect(mockCreateOverseasEntity).toHaveBeenCalledTimes(1);
       expect(mockUpdateOverseasEntity).not.toHaveBeenCalled();
 
       expect(resp.text).toContain(`${FOUND_REDIRECT_TO} ${PRESENTER_URL}`);
     });
 
+    test(`redirect to the ${PRESENTER_PAGE} page after a successful update of transaction and overseas entity with REDIS_removal flag set to ON`, async () => {
+      mockIsActiveFeature.mockReturnValueOnce(true); // For FEATURE_FLAG_ENABLE_SAVE_AND_RESUME
+      mockIsActiveFeature.mockReturnValueOnce(true); // For FEATURE_FLAG_ENABLE_REDIS_REMOVAL
+
+      const mockData = { ...APPLICATION_DATA_MOCK, [Transactionkey]: null, [OverseasEntityKey]: null };
+      mockGetApplicationData.mockReturnValueOnce(mockData);
+      const resp = await request(app).post(OVERSEAS_NAME_URL).send({ [EntityNameKey]: OVERSEAS_NAME_MOCK });
+
+      expect(resp.status).toEqual(302);
+      expect(mockData[Transactionkey]).toEqual(null);
+      expect(mockData[OverseasEntityKey]).toEqual(null);
```

Comment:
same here for the TRANSACTION_ID and OVERSEAS_ENTITY_ID check

Reply:
Done.